src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -5,3 +5,85 @@ title: Motion
+
+`ExecuteScriptByName` takes
+
+- a subsumption ID (subsumption is a priority for the servos)
+- the name of the script/s as a string or a vector of strings
+- the duration of the scripts
+- the time that these scripts should start executing
+
+An example of receiving this message is in [ScriptEngine.cpp](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/module/motion/ScriptEngine/src/ScriptEngine.cpp#L51).
+
+```cpp
+on<Trigger<ExecuteScriptByName>>().then([this](const ExecuteScriptByName& command) {...}
+```
+
+An example of emitting this message is in [KickScript.cpp](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/module/behaviour/skills/KickScript/src/KickScript.cpp#L102)
+
+```cpp
+emit(std::make_unique<ExecuteScriptByName>(
+    id, std::vector<std::string>({"Stand.yaml", "KickRight.yaml", "Stand.yaml"})));
+```
+
+Note these are only the names of the scripts, not the full paths. The names correspond to the names of YAML files in the [scripts folder](https://github.com/NUbots/NUbots/tree/master/module/motion/ScriptEngine/data/scripts).
+
+In addition to using the name of a script, we can also use [an instance of Script](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/shared/extension/Script.h#L43) directly. `Script` can be instantiated with all the information for the script to execute, and can be emitted using the `ExecuteScript` message. `ExecuteScript` takes:
+
+- a subsumption ID (subsumption is a priority for the servos)
+- a Script or vector of Scripts
+- the duration of the scripts
+- the time that these scripts should start executing
+
+An example of receiving this messages is in [ScriptEngine.cpp](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/module/motion/ScriptEngine/src/ScriptEngine.cpp#L71)
+
+```cpp
+on<Trigger<ExecuteScript>>().then([this](const ExecuteScript& command) {...}
+```
+
+An example of emitting this message is in [ScriptTuner.cpp](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/module/behaviour/tools/ScriptTuner/src/ScriptTuner.cpp#L590)
+
+```cpp
+emit(std::make_unique<ExecuteScript>(id, script, NUClear::clock::now()));
+```
+
+The [ScriptEngine](https://github.com/NUbots/NUbots/tree/master/module/motion/ScriptEngine) module receives either of these messages and processes the information for the script. It collects the servo positions and time for each frame and modifies them if needed based on the given duration and start time in the message. The information is then emitted as a [ServoCommand](https://github.com/NUbots/NUbots/blob/master/shared/message/behaviour/ServoCommand.proto) message.
+
+## Script files
+
+The script files are YAML files specifying a duration and list of servo targets, in a format like the following:
```

Comment:
Can we make this example illustrative of how this should be used - by this I mean, can we clarify if `gain` is always 30 like in this example, and `torque` is always 100? This example makes it seem like yes to both of those.

I think if we labeled the units with yaml comments it would also be good, because then we could show off what comments look like in the example too, as well as encourage commenting in the yaml files

Reply:
Those are the usual/common values I think, but they won't always be 30 or 100.

Reply:
I think that's knowledge that we could pass on with these docs, and which isn't otherwise intuitive.